### DISCUSSION TITLE
RotaryEncoder - SecondFunc

### DISCUSSION THREAD

**Discussion by oxygi** [MiczFlor/RPi-Jukebox-RFID#2247] [Q&A]
Before I open an issue, I wanted to ask the community. I have a RotaryEncoder and use this code for the click: 

[RotaryEncoder_Press]
type : Button
pin : 13
pull_up_down : pull_up
functioncall : functionCallPlayerNext
hold_mode: SecondFunc
hold_time: 2.0
functionCall2: functionCallPlayerSeekFwd

[RotaryEncoder_Rotate]
type : RotaryEncoder
timebase : 0.02
pin1 : 5
pin2 : 6
functioncall1 : functionCallVolU
functioncall2 : functionCallVolD

Now, if I press the button for a long time, it immediately switches to the next song instead of fast-forwarding. I have already tried several parameters. Can anyone help me here?

I am on Version [v2.5.0]
Thank you very much in advance!

**Comment by AlvinSchiller** [comment]
Hi, this seems to be the same case as https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/1666.
Unfortunately there is currently no other way to solve this behaviour.

**Reply by AlvinSchiller** [reply]
https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/1666 has been updated with a possible solution as local patch.

**Reply by AlvinSchiller** [reply]
This is now fixed with https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2367 for v2.8.0.
The behavior of `hold_mode` `SecondFunc` and `SecondFuncRepeat` has changed. The secondary function is no longer triggered additionally to the primary function.
Now its called exclusively if `hold_time` is reached. The primary function will only be triggered if the button is pressed shorter then `hold_time`! Existing configurations may need to adapt to this.

If you still encounter the execution of `functioncall ` after the hold, try setting the `antibouncehack` to `True`.
